protocols/ipfs.ts

```diff
@@ -1,151 +1,86 @@
-
-        const ipfsOptions = {
-          repo: this.options.path,
-          config: {
-            Experimental: {
-              AcceleratedDHTClient: true
-            },
-            Gateway: null,
-            Addresses: {
-              API: `/ip4/127.0.0.1/tcp/${apiPort}`,
-              Gateway: null,
-              Swarm: [
-                `/ip4/0.0.0.0/tcp/${swarmPort}`,
-                `/ip6/::/tcp/${swarmPort}`,
-                `/ip4/0.0.0.0/udp/${swarmPort}/quic`,
-                `/ip6/::/udp/${swarmPort}/quic`
-              ]
-            },
-            Ipns: {
-              UsePubSub: true
-            },
-            PubSub: {
-              Enabled: true
-            },
-            Swarm: {
-              ConnMgr: {
-                HighWater: 512
-              }
-            }
-          }
-        }
-        const ipfsdOpts: ControllerOptions = {
-          type: 'go',
-          ipfsOptions,
-          ipfsHttpModule: IPFSHTTPClient,
-          ipfsBin: GoIPFS.path()
-        }
-
-        let ipfsd: Controller | null = null
-
-        try {
-          ipfsd = await createController(ipfsdOpts)
-          await ipfsd.init()
-          await ipfsd.start()
-          await ipfsd.api.id()
-        } catch (cause) {
-          const { repo } = ipfsOptions
-          const lockFile = path.join(repo, 'repo.lock')
-          const apiFile = path.join(repo, 'api')
-          try {
-            await Promise.all([
-              rm(lockFile),
-              rm(apiFile)
-            ])
-            ipfsd = await createController(ipfsdOpts)
-            await ipfsd.start()
-            await ipfsd.api.id()
-          } catch (cause) {
-            const message = 'Unable to start IPFS daemon'
-            throw createError(500, message, { cause })
-          }
-        }
-
-        let gracefulStop = false
-
-        // When launching kubo, account for early exits
-        if (ipfsd.subprocess != null) {
-          void ipfsd.subprocess.once('exit', () => {
-            if (!gracefulStop) {
-              this.load().catch((e) => {
-                console.error('Unable to restart kubo after it died.')
-                console.error(e.stack)
-                process.exit(1)
-              })
-            }
-          })
-        }
-
-        this.ipfs = ipfsd.api
-        this.onCleanup.push(async () => {
-          gracefulStop = true
-          if (ipfsd !== null) {
-            await ipfsd.stop()
-          }
-        })
-      } else if (this.options.provider === KUBO) {
-        // rpcURL is for connecting to a Kubo Go-IPFS node
-        this.ipfs = IPFSHTTPClient.create({
-          url: this.options.path
-        })
+    const datastorePath = path.join(this.options.path, 'datastore')
+    const blockstorePath = path.join(this.options.path, 'blockstore')
+    const datastore = new FsDatastore(datastorePath)
+    const blockstore = new FsBlockstore(blockstorePath)
+
+    const libp2pConfig = {
```

Comment:
Does this run exclusively in Node.js?
Will the Helia node be providing data for retrieval from the web? You probably want AutoTLS too (which is enabled in the defaults)

Any reason you are starting with custom libp2p config?

I would recommend using the exported libp2p defaults function or just https://github.com/ipfs/helia/blob/main/packages/helia/src/utils/libp2p-defaults.ts
https://ipfs.github.io/helia/functions/helia.libp2pDefaults.html